lib/rubocop/cop/style/struct_inheritance.rb

```diff
@@ -0,0 +1,40 @@
+# encoding: utf-8
+
+module RuboCop
+  module Cop
+    module Style
+      # This cop checks for inheritance from Struct.new
+      #
+      # @example
+      #
+      # # bad
+      # class Person < Struct.new(:first_name, :last_name)
+      #
+      # # good
+      # Person = Struct.new(:first_name, :last_name) do
+      class StructInheritance < Cop
+        MSG = "Don't extend a Struct.new - it already is a new class."
+
+        def on_class(node)
+          _name, superclass, _body = *node
+          if struct_construtor?(superclass)
+            add_offense(node, superclass.loc.expression, MSG)
+          end
+        end
```

Comment:
As travis says, when running Rubocop against this file:
```
lib/rubocop/cop/style/struct_inheritance.rb:20:11: C: Use a guard clause instead of wrapping the code inside a conditional expression.
          if struct_construtor?(superclass)
          ^^
```

So something like:
```rb
def on_class(node)
  _name, superclass, _body = *node
  return unless struct_construtor?(superclass)

  add_offense(node, superclass.loc.expression, MSG)
end
```